pydantic/function_model.py

```diff
@@ -0,0 +1,442 @@
+        """
+        parsed, metadata = self.validate_arguments(*args, **kwargs)
+
+        self._parsed = parsed
+        self._parameters = metadata
+
+    def update_argument(self, pos_or_kw: Union[int, str], value: Any) -> None:
+        """Update an initialized argument.
+
+        Args:
+            pos_or_kw: A position or keyword that identifies the intended parameter.
+            value: The value to set the argument to.
+        """
+        if self._parsed is None:
+            raise AttributeError('Arguments have not been initialized, so they cannot be updated.')
+
+        try:
+            key, details = self._get_parameter(pos_or_kw)
+        except (PydanticKnownError, PydanticCustomError) as error:
+            if isinstance(error, PydanticKnownError):
+                error = error.type
+
+            init: InitErrorDetails = {
+                'type': error,
+                'loc': (pos_or_kw,),
+                'input': value,
+            }
+            raise ValidationError.from_exception_data(self.__class__.__name__, [init])
+
+        if key == 'args':
+            assert isinstance(pos_or_kw, int)  # Should always be true, otherwise something is broken
+
+            args_pos = pos_or_kw - self._positional_total
+            args = getattr(self._parsed, 'args')
+            args_len = len(args)
+            if args_len <= args_pos - 1:
+                # The previous positional argument hasn't been set yet, so we can't set this one
+                raise AttributeError(
+                    f'Cannot set argument at position {pos_or_kw}, as the preceding argument has not been set.'
+                )
+            elif args_len <= args_pos:
+                # This argument hasn't been set yet, but the preceding one has, so append it
+                args.append(value)
+            else:
+                args[args_pos] = value
+
+        else:
+            setattr(self._parsed, key, value)
+
+    def validate_argument(self, pos_or_kw: Union[int, str], value: Any) -> Any:
+        """Validate an argument against a specific parameter.
+
+        Args:
+            pos_or_kw: A position or keyword that identifies the intended parameter.
+            value: The value to validate as a valid parameter input.
+        """
+        try:
+            key, details = self._get_parameter(pos_or_kw)
+        except (PydanticCustomError, PydanticKnownError) as error:
+            if isinstance(error, PydanticKnownError):
+                error = error.type
+
+            init: InitErrorDetails = {
+                'type': error,
+                'loc': (pos_or_kw,),
+                'input': value,
+            }
+            raise ValidationError.from_exception_data(self.__class__.__name__, [init])
+
+        if key == 'args':
+            # Values within *args are unrestricted
+            return value
+        elif key == 'kwargs':
+            # Values within **kwargs are unrestricted
+            return value
```

Comment:
```suggestion
        if key in {'args', 'kwargs'}:
            # Values within *args and **kwargs are unrestricted
            return value
```

Reply:
Fixed, I believe at some point I had it where the two keys were doing something different and I just forgot to reduce it down.